dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeAPIImpl.java

```diff
@@ -294,15 +295,19 @@ public ContentType findDefault() throws DotDataException, DotSecurityException {
     }
   }
 
-  @Override
-  public List<SimpleStructureURLMap> findStructureURLMapPatterns() throws DotDataException {
-    List<SimpleStructureURLMap> res = new ArrayList<SimpleStructureURLMap>();
+    @Override
+    public List<SimpleStructureURLMap> findStructureURLMapPatterns() throws DotDataException {
+        List<SimpleStructureURLMap> res = new ArrayList<>();
```

Comment:
I think you can create a builder and add the elements to it, and build at the end a immutable list instead of add to the ArrayList and the make a copy. Just a comment